server/protocols/jmap-draft/src/test/java/org/apache/james/jmap/event/PreComputeMessagePreviewListenerTest.java

```diff
@@ -0,0 +1,181 @@
+    private static final MailboxPath BOB_INBOX_PATH = MailboxPath.inbox(BOB);
+    private static final MailboxPath BOB_OTHER_BOX_PATH = MailboxPath.forUser(BOB, "otherBox");
+
+    private MessagePreviewStore messagePreviewStore;
+    private MailboxSession mailboxSession;
+    private StoreMailboxManager mailboxManager;
+
+    private MessageManager inboxMessageManager;
+    private MessageManager otherBoxMessageManager;
+
+    @BeforeEach
+    void setup() throws Exception {
+        InMemoryIntegrationResources resources = InMemoryIntegrationResources.defaultResources();
+        mailboxManager = resources.getMailboxManager();
+        MessageIdManager messageIdManager = resources.getMessageIdManager();
+
+        mailboxSession = MailboxSessionUtil.create(BOB);
+
+        MailboxId inboxId = mailboxManager.createMailbox(BOB_INBOX_PATH, mailboxSession).get();
+        inboxMessageManager = mailboxManager.getMailbox(inboxId, mailboxSession);
+
+        MailboxId otherBoxId = mailboxManager.createMailbox(BOB_OTHER_BOX_PATH, mailboxSession).get();
+        otherBoxMessageManager = mailboxManager.getMailbox(otherBoxId, mailboxSession);
+
+        messagePreviewStore = new MemoryMessagePreviewStore();
+
+        HtmlTextExtractor htmlTextExtractor = new JsoupHtmlTextExtractor();
+        MessageContentExtractor messageContentExtractor = new MessageContentExtractor();
+
+        MessageFullViewFactory messageFullViewFactory = new MessageFullViewFactory(resources.getBlobManager(), messageContentExtractor, htmlTextExtractor);
+
+        PreComputeMessagePreviewListener listener = new PreComputeMessagePreviewListener(mailboxManager, messageIdManager, messagePreviewStore, messageFullViewFactory);
+        resources.getEventBus().register(listener);
+    }
+
+    @Test
+    void deserializeMailboxAnnotationListenerGroup() throws Exception {
+        assertThat(Group.deserialize("org.apache.james.jmap.event.PreComputeMessagePreviewListener$PreComputeMessagePreviewListenerGroup"))
+            .isEqualTo(new PreComputeMessagePreviewListener.PreComputeMessagePreviewListenerGroup());
+    }
+
+    @Test
+    void shouldStorePreviewWhenBodyMessageNotEmpty() throws Exception {
+        ComposedMessageId composedId = inboxMessageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .build(previewMessage()),
+            mailboxSession);
+
+        assertThat(Mono.from(messagePreviewStore.retrieve(composedId.getMessageId())).block())
+            .isEqualTo(PREVIEW);
+    }
+
+    @Test
+    void shouldStoreEmptyPreviewWhenEmptyBodyMessage() throws Exception {
+        ComposedMessageId composedId = inboxMessageManager.appendMessage(
+            MessageManager.AppendCommand.builder()
+                .build(emptyMessage()),
+            mailboxSession);
+
+        assertThat(Mono.from(messagePreviewStore.retrieve(composedId.getMessageId())).block())
+            .isEqualTo(Preview.NO_BODY);
```

Comment:
does that mean we store "(Empty)" in the underlying storage? or do we store a null object?

Reply:
"Empty" for now, but I believe it joins more in details in this conversation : https://github.com/linagora/james-project/pull/2918#discussion_r350829600

I get your concern and I agree, however it's not related to this PR right?

Reply:
With @trantienduchn changes, we are going to store `""` not `(Empty)` so that fixing the locale bug won't require a full recompute.